### PR TITLE
cat: validate types, formats, airbyte types and their combinations on catalog

### PR DESCRIPTION
**Pull request by alafanechere**
## What
Closes https://github.com/airbytehq/airbyte-internal-issues/issues/1342

We want to make sure: 
- the types declared in stream properties are supported.
- the formats declared in stream properties are supported.
- the airbyte types declared in stream properties are supported.
- the combination of types and formats are supported.
- the combination of types and airbyte types are supported.

## How
* Add a new `test_catalog_has_supported_data_types` test to `test_core.py` on CAT 
* Unit test it

### CODE REVIEW

**Review by evantahler — APPROVED** [review]

**Comment by evantahler** [comment]
cc @edgao Anything else we should test here?

File: airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/tests/test_core.py [hunk 1]
```diff
@@ -566,6 +566,22 @@ def test_check(self, connector_config, inputs: ConnectionTestConfig, docker_runn
...
+        ({"string"}, "date"),
+        ({"string"}, "date-time"),
+        ({"string", "null"}, "date"),
+        ({"string", "null"}, "date-time"),
+    ]
+    VALID_TYPE_AIRBYTE_TYPE_COMBINATIONS = [
+        ({"string"}, "timestamp_with_timezone"),
+        ({"string"}, "timestamp_without_timezone"),
+        ({"string", "null"}, "timestamp_with_timezone"),
+    ]
```
[the remark this answers is not in the document]
Comment (alafanechere) [reply airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/tests/test_core.py:583]: I think we would not need this test at all if it was in the protocol as deserialization of the streams would likely fail upstream when we run `discover`.

**Pull request merged**